cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/sqlite/SQLDatabaseFactory.java

```diff
@@ -89,7 +90,9 @@ private static boolean isFtsAvailable() {
     public static SQLDatabase openSQLDatabase(File dbFile, KeyProvider provider) throws IOException, SQLException {
         Misc.checkNotNull(dbFile, "dbFile");
         File dbDirectory = dbFile.getParentFile();
-        dbDirectory.mkdirs();
+        if (!dbDirectory.mkdirs()){
+            logger.info(String.format(Locale.ENGLISH, "Did not create directories for path: %s directories may already exist", dbFile));
```

Comment:
Not sure if this deserves a logger.info. The more interesting case is probably when it does create the directory.